### ISSUE TITLE
Feature request: Saved scan list

### ISSUE THREAD

**Issue by nhfo** [markusfisch/BinaryEye#50]
Please add an option to save barcodes. As such there is the history list, but since there, entries do not have title and/or description it can become difficult to find perticular scan as number of enteries grow. It would be practical to have another list for saved scans, with option for performing search.

My primary use case is to scan barcodes etc. and display those in the original format when needed. And saved scan list would improve apps usability many folds.

**Comment by markusfisch** [comment]
Yes, I see. The current history is just a listing you may export and clear afterwards. But the barcodes are already saved in a Sqlite database so it shouldn't be too hard to add features like searching or something.

While the original barcode of a history entry isn't saved, you can recreate the barcode from the history entry by selecting it and using the `+` action icon in the top right corner. Of course, that's just a workaround 😉

**Comment by nhfo** [comment]
> While the original barcode of a history entry isn't saved, you can recreate the barcode from the history entry by selecting it and using the + action icon in the top right corner. Of course, that's just a workaround 😉

Indeed, that's how I make use of it now, although finding right scan just by looking at scan data which is in most instances just random digits is somewhat hard, hence title and/or description fields in list entries and search option would be helpful.

**Comment by nhfo** [comment]
@markusfisch this is one feature I really want. Are you planning to implement it? I would really appreciate if it is taken up. :)

**Comment by markusfisch** [comment]
Yes, it's still on the list! Please bear with me 😉 I already made some small improvements to the history listing lately and I will add searching it in the near future. Also, I'm wondering if just adding tags (one or more words) to a scan (instead of a title and/or description) would fit your application too? Or would you rather like to have title/description for each scan?

**Comment by nhfo** [comment]
Tag approach would do. I scan all sorts of loyalty cards, which are usually just string of numbers and therefore some means to name them is essential.

Although having another field for longer description/note would be useful in some cases, support for it can perhaps be deferred.

**Comment by nlsdg** [comment]
Shouldn't this request be closed? Scan history with support for naming scans has been implemented.
